salt/modules/virt.py

```diff
@@ -6796,19 +7340,287 @@ def network_define(name, bridge, forward, ipv4_config=None, ipv6_config=None, **
         conn.close()
         return False
 
-    if (starting is True or autostart is True) and network.isActive() != 1:
+    if (start or autostart) and network.isActive() != 1:
         network.create()
 
-    if autostart is True and network.autostart() != 1:
+    if autostart and network.autostart() != 1:
         network.setAutostart(int(autostart))
-    elif autostart is False and network.autostart() == 1:
+    elif not autostart and network.autostart() == 1:
         network.setAutostart(int(autostart))
 
     conn.close()
 
     return True
 
 
+def _remove_empty_xml_node(node):
+    """
+    Remove the nodes with no children, no text and no attribute
+    """
+
+    def empty_node_remover(node):
+        for child in node:
+            if not child.tail and not child.text and not child.items() and not child:
+                node.remove(child)
+
+    return xmlutil.visit(node, empty_node_remover)
```

Comment:
**suggestion**: something feels off about creating an internal function to allow recursion. This function could also directly recurse:

```
def _remove_empty_xml_node(node):
    """
    Remove the nodes with no children, no text and no attribute
    """
     for child in node:
         if not child.tail and not child.text and not child.items() and not child:
             node.remove(child)
         else:
             _remove_empty_xml_node(child)

    return node
   ```

Reply:
:+1: